server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlExecParser.g4

```diff
@@ -1767,82 +1704,114 @@ empty_rule: /* epsilon */;
+dbs_parameter: dbs_expression | TABLE dbs_sql_identifier | NULL;
+//dbs_package_name:  dbs_sql_identifier DOT_FS (dbs_sql_identifier | ASTERISKCHAR); // added below correction
+dbs_package_name:  (dbs_sql_identifier DOT_FS)? dbs_sql_identifier;
+dbs_sequence_name : dbs_sql_identifier (DOT_FS dbs_sql_identifier)?; //  {validateLength($T.text, "Sequence name", 128);};
+dbs_trigger_name :  dbs_sql_identifier (DOT_FS dbs_sql_identifier)?; // {validateLength($T.text, "Trigger name", 128);};
+dbs_version_name: CHAR_STRING_LITERAL | dbs_host_variable | ;//empty alternative;
+dbs_object_name: dbs_sql_identifier (DOT_FS dbs_sql_identifier)?;
+dbs_sql_variable_reference: dbs_host_variable | dbs_object_name;
+dbs_statement_name: dbs_sql_identifier;
+dbs_xquery_context_item_expression: dbs_generic_name;
+/////  validation rules /////
+dbs_integer5: T=INTEGERLITERAL  {validateValue($T.text, "5");};
+dbs_integer12: T=INTEGERLITERAL  {validateValue($T.text, "12");};
+dbs_integer1200: T=INTEGERLITERAL  {validateValue($T.text, "1200");};
+dbs_integer1208: T=INTEGERLITERAL  {validateValue($T.text, "1208");};
+dbs_decfloat_integer: INTEGERLITERAL {validate34or16($INTEGERLITERAL.text);};
+dbs_decimal_15_31: INTEGERLITERAL {validateTokenWithRegex($INTEGERLITERAL.text, "\\b(15|31)\\b", "15 or 31 are only allowed");};
+dbs_pagenum_char_a_r: IDENTIFIER  {validateTokenWithRegex($IDENTIFIER.text, "^[aArR]$", "unknown token. Expected A, R");};
+dbs_char_s: T=IDENTIFIER {validateValue($T.text, "S");};
+dbs_function_language: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)\\b(ASSEMBLE|C|COBOL|JAVA|PLI)\\b", "unknown token. Supported tokens are JAVA, ASSEMBLE, C, COBOL, PLI");};
+dbs_function_parameter_style: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)\\b(SQL|JAVA)\\b", "unknown token. Supported tokens are JAVA, SQL");};
+dbs_procedure_language: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)\\b(ASSEMBLE|C|COBOL|JAVA|PLI|REXX)\\b", "unknown token. Supported tokens are JAVA, ASSEMBLE, C, COBOL, PLI, REXX");};
+oneof_lang: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)\\b(ASSEMBLE|C|COBOL|JAVA|PLI|REXX|SQL)\\b", "unknown token. Supported tokens are JAVA, ASSEMBLE, C, COBOL, PLI, REXX, SQL");};
+dbs_exact_match_identifier_sql: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)\\b(SQL)\\b", "unknown token. Supported tokens is SQL");};
+dbs_k_m_g_identifier: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)\\b(K|M|G)\\b", "unknown token. Supported tokens is K, M, G");};
+kmg_blob_parameter: INTEGERLITERAL dbs_k_m_g_identifier? | IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "(?i)^\\d+[KMG]?$", "unknown token");};
+dbs_g_char_identifier: IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "^[gG]$", "unknown token. Expected G");};
+dbs_dsize_parameter: INTEGERLITERAL dbs_g_char_identifier | IDENTIFIER {validateTokenWithRegex($IDENTIFIER.text, "\\d+[Gg]", "unknown token");};
```

Comment:
The doc additionally says `Specify a power-of-two integer in the range 1–256`.

Reply:
added to visitor.